src/lightning/pytorch/callbacks/gradient_accumulation_scheduler.py

```diff
@@ -99,5 +98,44 @@ def get_accumulate_grad_batches(self, epoch: int) -> int:
                 break
         return accumulate_grad_batches
 
+    def on_train_start(self, trainer: "pl.Trainer", pl_module: "pl.LightningModule") -> None:
```

Comment:
The validation of settings now happens here instead of Trainer's config validator so that the entire feature is self-contained in the callback.